main/solution/backend/config/infra/cloudformation.yml

```diff
@@ -307,9 +307,158 @@ Resources:
+          - Effect: Allow
+            Action:
+              - storagegateway:CreateNFSFileShare
+              - storagegateway:ListFileShares
+              - storagegateway:DescribeNFSFileShares
+              - storagegateway:UpdateNFSFileShare
+            Resource: !Sub 'arn:aws:storagegateway:${AWS::Region}:${AWS::AccountId}:gateway/*'
+          - Effect: Allow
+            Action:
+              - states:StartExecution
+            Resource:
+              - !Ref SMWorkflow
+          - Effect: Allow
+            Action:
+              - s3:GetObject*
+              - s3:AbortMultipartUpload
+              - s3:ListMultipartUploadParts
+              - s3:PutObject*
+              - s3:DeleteObject*
+            Resource:
+              - 'arn:aws:s3:::${self:custom.settings.studyDataBucketName}/*'
+              - 'arn:aws:s3:::${self:custom.settings.environmentsBootstrapBucketName}/*'
+              - 'arn:aws:s3:::${self:custom.settings.externalCfnTemplatesBucketName}/*'
+              - 'arn:aws:s3:::${self:custom.settings.envTypeConfigsBucketName}/*'
+          - Effect: Allow
+            Action:
+              - s3:ListBucket
+              - s3:GetBucketPolicy
+              - s3:PutBucketPolicy
+            Resource:
+              - 'arn:aws:s3:::${self:custom.settings.studyDataBucketName}'
+              - 'arn:aws:s3:::${self:custom.settings.environmentsBootstrapBucketName}'
+              - 'arn:aws:s3:::${self:custom.settings.envTypeConfigsBucketName}'
+          - Effect: Allow
+            Action:
+              - kms:GenerateDataKey
+              - kms:Decrypt
+              - kms:DescribeKey
+              - kms:Encrypt
+              - kms:ReEncrypt*
+              - kms:GetKeyPolicy
+              - kms:PutKeyPolicy
+            Resource:
+              # NOTE: Can't use '!GetAtt StudyDataEncryptionKey.Arn' due to
+              #   circular dependency in StudyDataEncryptionKey's KeyPolicy
+              - !Sub 'arn:aws:kms:${AWS::Region}:${AWS::AccountId}:alias/${self:custom.settings.studyDataKmsKeyAlias}'
+              - !Sub 'arn:aws:kms:${AWS::Region}:${AWS::AccountId}:key/*'
+          - Effect: Allow
+            Action:
+              - sagemaker:CreatePresignedNotebookInstanceUrl
+            Resource:
+              - !Sub 'arn:aws:sagemaker:${AWS::Region}:${AWS::AccountId}:notebook-instance/basicnotebookinstance-*'
+          - Effect: Allow
+            Action:
+              - ec2:DescribeInstanceStatus
+              - ec2:DescribeInstances
+            Resource: '*' # For the actions listed above IAM does not support resource-level permissions and requires all resources to be chosen
+          - Effect: Allow
+            Action:
+              - ec2:DescribeSubnets
+              - ec2:DescribeVpcs
+              - ec2:DescribeNetworkInterfaces
+            Resource: '*' # For the actions listed above IAM does not support resource-level permissions and requires all resources to be chosen
```

Comment:
Where do we manage EC2 resources in main account?

Reply:
Just AMIs and their configurations, so this are only required in workflow look runner.